multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -614,39 +619,33 @@ def _fit(
         hpo_mode: bool = False,
         **hpo_kwargs,
     ):
-        if presets is None:
-            presets = "siamese_network"
-
-        if presets == "siamese_network":
-            config = self._config
-            config = get_config(
-                problem_type=self._pipeline,
-                presets=presets,
-                config=config,
-                overrides=hyperparameters,
-                extra=["matcher"],
+        config = self._config
+        config = get_config(
+            problem_type=self._pipeline,
+            presets=presets,
+            config=config,
+            overrides=hyperparameters,
+            extra=["matcher"],
+        )
+
+        if self._query_config is None:
+            query_config = copy.deepcopy(config)
```

Comment:
avoid deepcopy? https://github.com/autogluon/autogluon/pull/2729

Reply:
This is to make sure the query_config and response_config are completely detached. copy.copy may not work since there are objects, e.g., list, in the config values. Since this is in training, one time deepcopy doesn't affect much.